tests/lib/managers/folders-test.js

```diff
@@ -276,4 +276,20 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('deletePermanently()', function() {
+
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').returns(done);
+			sandbox.stub(boxClientFake, 'del').withArgs('/folders/' + FOLDER_ID + '/trash').yieldsAsync();
+			folders.deletePermanently(FOLDER_ID, done);
+		});
+
+		it('should make DELETE call to remove folder permanently', function(done) {
+
+			sandbox.stub(boxClientFake, 'defaultResponseHandler').yields();
+			sandbox.mock(boxClientFake).expects('del').withArgs('/folders/' + FOLDER_ID + '/trash', null);
```

Comment:
This stub/mock combination isn't quite right; `.yields()` means that `defaultResponseHandler()` will call its function argument immediately when called.  In this test, that argument is `done`, so the test will end immediately on that line.  This is incorrect; `defaultResponseHandler()` should instead return a function and `boxClient.del()` should yield asynchronously to it.  See your above test.